classes/form/guestlink_access_form.php

```diff
@@ -0,0 +1,104 @@
+ */
+class guestlink_access_form extends \moodleform {
+
+    /**
+     * Define (add) particular settings this activity can have.
+     *
+     * @return void
+     */
+    public function definition() {
+        global $CFG, $DB, $OUTPUT, $PAGE;
+        $mform = &$this->_form;
+
+        // Guest Access URL.
+        $inviteparticipantselement = $mform->createElement('text', 'guestlinkurl', get_string('view_guestlink_label', 'bigbluebuttonbn'), ['readonly' => true, 'size' => 60]);
+        $mform->setType('guestlinkurl', PARAM_TEXT);
+        $mform->setDefault('guestlinkurl', $this->_customdata['guestlink']['url']);
+        $inviteparticipantsgroup = [];
+        $inviteparticipantsgroup[] =& $inviteparticipantselement;
+        $inviteparticipantsgroup[] =& $mform->createElement('button', 'guestlinkurl_copy', get_string('mod_guestlink_access_form_copy', 'bigbluebuttonbn'));
+        $mform->addGroup($inviteparticipantsgroup, 'inviteparticipantsgroup', get_string('view_guestlink_label', 'bigbluebuttonbn'), array(' '), false);
+
+        // Access Code.
+        $accesscodeelement = $mform->createElement('text', 'password', '', [
+            'readonly' => true,
+            'size' => 29,
+            'placeholder' => get_string('view_guestlink_password_no_password_set', 'bigbluebuttonbn')
+        ]);
+        $mform->setType('password', PARAM_TEXT);
+        $mform->setDefault('password', $this->_customdata['guestlink']['password'] ?? '');
+        $accesscodegroup = [];
+        $accesscodegroup[] =& $accesscodeelement;
+        if (!empty($this->_customdata['guestlink']['changepassenabled'])) {
+            $accesscodegroup[] =& $mform->createElement('button', 'password_clear', get_string('mod_guestlink_access_form_clear', 'bigbluebuttonbn'));
+        }
+        if (!empty($this->_customdata['guestlink']['changepassenabled'])) {
+            $accesscodegroup[] =& $mform->createElement('button', 'password_change', get_string('mod_guestlink_access_form_generate', 'bigbluebuttonbn'));
+        }
+        $accesscodegroup[] =& $mform->createElement('button', 'password_copy', get_string('mod_guestlink_access_form_copy', 'bigbluebuttonbn'));
+        $mform->addGroup($accesscodegroup, 'accesscodegroup', get_string('view_guestlink_password_label', 'bigbluebuttonbn'), array(' '), false);
+
+        // Expires At.
+        $defaultdateoptions = [
+           'optional' => true
+        ];
+        $expiresatelement = $mform->createElement('date_time_selector', 'guestlinkexpiresat', '', $defaultdateoptions);
+        $mform->setType('guestlinkexpiresat', PARAM_RAW);
+        $accesscodegroup = [];
+        $accesscodegroup[] =& $expiresatelement;
+        $mform->addGroup($accesscodegroup, 'accesscodegroup', get_string('view_guestlink_expires_at_label', 'bigbluebuttonbn'), array(' '), false);
+
+        $this->add_action_buttons($cancel = false, get_string('view_guestlink_save_settings', 'bigbluebuttonbn'));
+
+        // CM ID.
```

Comment:
Are these elements (CMID, BBB ID) actually required?

Reply:
Yeah, I remember it submits to a particular endpoint that checks and verifies that those keys are supplied when submitted. But will recheck.

Reply:
I removed these fields in the recent commit